bigframes/core/tree_properties.py

```diff
@@ -25,3 +29,12 @@ def is_trivially_executable(node: nodes.BigFrameNode) -> bool:
 
 def local_only(node: nodes.BigFrameNode) -> bool:
     return all(isinstance(node, nodes.ReadLocalNode) for node in node.roots)
+
+
+@functools.cache
```

Comment:
Would this get unwieldy with local data? I worry about that Feather data getting stored forever.

Reply:
Yeah, not sure why I put that cache decorator. Would take a really long time for the cache to get too big though. Removed and will figure out the whole property caching thing later.